api/storage_service.cc

```diff
@@ -510,6 +510,54 @@ void set_storage_service(http_context& ctx, routes& r, sharded<service::storage_
         });
     });
 
+    ctx.db.local().get_builtin_routine_registry().register_routine("toppartitions",
+        db::make_single_value_routine<sstring>("toppartitions", [&ctx] (db::builtin_routine::parameter_map&& par) {
+            bool filters_provided = false;
+
+            std::unordered_set<std::tuple<sstring, sstring>, utils::tuple_hash> table_filters {};
+            if (par.contains("table_filters")) {
+                filters_provided = true;
+                auto filters = par["table_filters"];
+                std::stringstream ss { filters };
```

Comment:
`filters` can be moved-into (but please mind the comment just below)

Reply:
also, `std::ostringstream` should be enough

Reply:
I'm assuming you mean `std::istringstream`

Reply:
Yes, absolutely. My bad